projects/project3/src/main/scala/DataStructures/DatasetReader.java

```diff
@@ -0,0 +1,121 @@
+package DataStructures;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+
+import java.util.Set;
+import java.util.List;
+import scala.collection.JavaConversions;
+import scala.collection.Seq;
+
+
+public class DatasetReader {
+
+
+    public static Seq<EntityProfile> readDataset(String path) {
```

Comment:
I see that the reading is (still) sequential. As Spark operates in Scala (JVM), would have been better if you could use `spark.textFile()` for reading the data and then apply Object manipulation using Attribute class.